astropy/io/fits/hdu/compressed.py

```diff
@@ -666,6 +668,28 @@ def __init__(self, data=None, header=None, name=None,
         self._orig_bscale = self._bscale
         self._orig_bitpix = self._bitpix
 
+    @property
+    def name(self):
```

Comment:
Looking at it I think we might need a special setter as well as getter. The issue is that with the current implementation on main you can do (on a compressed header):

```python
>>> hdu.name = "FOO"
>>> hdu.header["EXTNAME"] = "BAR"
```
and when you do `hdu.writeto("new.fits")` you end up with "FOO" in the header and "BAR" is dropped.  I think this is because the name setter sets `._header` but the `hdu.header` is `._image_header`. I think this needs to be changed to make `.name` use `_image_header`.

Reply:
I changed it to use `self.header` in both places. That should cause it to sync to both `self._image_header` and `self._header` and so be consistent.